libs/domain/site/src/services/store/store.interceptor.ts

```diff
@@ -41,13 +41,11 @@ export class StoreInterceptor implements HttpInterceptor {
     options: RequestOptions,
     store: string
   ): RequestOptions {
-    return {
-      ...options,
-      headers: {
-        ...options.headers,
-        'X-Store': store,
-      } as HeadersInit,
-    };
+    const headers = new Headers(options.headers);
+
+    headers.set('X-Store', store);
```

Comment:
Can we change this to `Store` , according to AA on multi-store, this should already be the new standard. 

Also, we might need to make this optional, as it's only a thing in case of multi-store. Something for a follow up task I'd say.

Reply:
The whole store interceptor could maybe be renamed to `MultiStoreInterceptor`

Reply:
This task is not changing how store interceptor works - I only updated the way we work with headers as previously it was very fragile.
I would suggest creating another ticket for the interceptor changes.